glowing_waffles/photometry.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import print_function, division, absolute_import
+import numpy as np
+import math
+from astropy import units as u
+from astropy.stats import mad_std
+from astropy.io import fits
+from astropy.table import Table, join
+from photutils import aperture_photometry, CircularAperture, CircularAnnulus
+from coordinates import convert_pixel_wcs
+
+__all__ = ['photutils_stellar_photometry']
+
+def photutils_stellar_photometry(ccd_image, sources, aperture_radius, inner_annulus
+                                 outer_annulus):
+    """
+    Perform aperture photometry on an image, with a few options for estimating
+    the local background from an annulus around the aperture.
+
+    Parameters
+    ----------
+    ccd_image : `~ccdproc.CCDData`
+        Image on which to perform aperture photometry.
+
+    sources : `~astropy.table.Table`
+        Table of extracted sources. Assumed to be the output of
+        `~photutils.daofind()` source extraction function.
+
+    aperture_radius : float
+        Radius of aperture(s) in pixels.
+
+    inner_annulus : float
+        Inner radius of the annulus in pixels.
+
+    outer_annulus : float
+        Outer radius of the annulus in pixels.
+
+    Returns
+    -------
+    phot_table : `~astropy.table.Table`
+        Astropy table with columns for flux, x/y coordinates of center,
+        RA/dec coordinates of center, sky background per pixel,
+        net flux, aperture and annulus radii used, and flux error.
+    """
+
+    #if coord_type is not pixel:
+        # /*****  Call Stefan's WCS function to convert to pixels *****/
+
+    #check that the outer radius is greater or equal the inner radius for annulus
+    if inner_annulus > outer_annulus:
+        raise ValueError("outer_annulus must be greater or equal inner_annulus")
+
+    #check that the annulus inner radius is greater or equal the aperture radius
+    if aperture_radius > inner_annulus:
+        raise ValueError("inner_radius must be greater or equal aperture_radius")
+
+    #Extract x,y coordinates from sources table, contstruct aperture and
+    #annulus objects from coordinates, and peform aperture photometry
+    coords = (sources['xcentroid'], sources['ycentroid'])
+    apertures = CircularAperture(coords, aperture_radius)
+    annulus = CircularAnnulus(coords, inner_annulus, outer_annulus)
+    phot_table = aperture_photometry(ccd_image, apertures)
+    phot_table_1 = aperture_photometry(ccd_image, annulus)
+
+    #Obtain the local background/pixel and net flux between the aperture and annulus objects
+    n_pix_ap = math.pi * (aperture_radius**2)
```

Comment:
Also, the apertures may have an `area` property (or method) that calculates this for you.